service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/request/EhrExtractAckHandler.java

```diff
@@ -0,0 +1,79 @@
+import uk.nhs.adaptors.gp2gp.ehr.EhrExtractStatusService;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus.EhrReceivedAcknowledgement;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus.EhrReceivedAcknowledgement.EhrReceivedAcknowledgementBuilder;
+import uk.nhs.adaptors.gp2gp.ehr.model.EhrExtractStatus.EhrReceivedAcknowledgement.ErrorDetails;
+import uk.nhs.adaptors.gp2gp.mhs.InvalidInboundMessageException;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class EhrExtractAckHandler {
+    private static final String ACK_TYPE_CODE_XPATH = "//MCCI_IN010000UK13/acknowledgement/@typeCode";
+    private static final String ACK_DETAILS_XPATH = "//MCCI_IN010000UK13/acknowledgement/acknowledgementDetail/code";
+    private static final String MESSAGE_REF_XPATH = "//MCCI_IN010000UK13/acknowledgement/messageRef/id/@root";
+    private static final String MESSAGE_ID_ROOT_XPATH = "//MCCI_IN010000UK13/id/@root";
+    private static final String ERROR_CODE_XPATH = "//justifyingDetectedIssueEvent/code";
+    private static final String ACK_OK_CODE = "AA";
+    private static final String ACK_BUSINESS_ERROR_CODE = "AE";
+    private static final String ACK_REJECTED_CODE = "AR";
+    private static final String CODE_ATTRIBUTE = "code";
+    private static final String DISPLAY_ATTRIBUTE = "displayName";
+
+
+    private final XPathService xPathService;
+    private final EhrExtractStatusService ehrExtractStatusService;
+
+    @SneakyThrows
+    public void handle(String conversationId, Document document) {
+        String ackTypeCode = xPathService.getNodeValue(document, ACK_TYPE_CODE_XPATH);
+        Instant now = Instant.now();
+        String messageRef = xPathService.getNodeValue(document, MESSAGE_REF_XPATH);
+        String rootId = xPathService.getNodeValue(document, MESSAGE_ID_ROOT_XPATH);
+        EhrReceivedAcknowledgementBuilder ackBuilder = EhrReceivedAcknowledgement.builder()
+                .messageRef(messageRef)
+                .rootId(rootId)
+                .received(now);
+
+        if (ACK_OK_CODE.equals(ackTypeCode)) {
+            LOGGER.info("Application Acknowledgement Accept ({}) received, messageRef: {}", messageRef, ackTypeCode);
+            ackBuilder.conversationClosed(now);
+        } else if (ACK_BUSINESS_ERROR_CODE.equals(ackTypeCode)) {
+            LOGGER.info("Application Acknowledgement Error ({}) received, messageRef: {}", messageRef, ackTypeCode);
+            ackBuilder.errors(extractErrorCodes(document, ERROR_CODE_XPATH));
+        } else if (ACK_REJECTED_CODE.equals(ackTypeCode)) {
+            LOGGER.info("Application Acknowledgement Reject ({}) received, messageRef: {}", messageRef, ackTypeCode);
+            ackBuilder.errors(extractErrorCodes(document, ACK_DETAILS_XPATH));
+        } else {
+            throw new InvalidInboundMessageException("Unsupported MCCI_IN010000UK13/acknowledgement/@typeCode: " + ackTypeCode);
```

Comment:
We could make use of the `ACK_TYPE_CODE_XPATH` constant here, in case it's value was to change